docs/pages/radiogroup.js

```diff
@@ -0,0 +1,695 @@
+// @flow strict
+import { type Node } from 'react';
+import { RadioGroup } from 'gestalt';
+import GeneratedPropTable from '../components/GeneratedPropTable.js';
+import PageHeader from '../components/PageHeader.js';
+import MainSection from '../components/MainSection.js';
+import Page from '../components/Page.js';
+import { multipledocgen, type DocGen } from '../components/docgen.js';
+
+export default function DocsPage({
+  generatedDocGen,
+}: {|
+  generatedDocGen: {| [string]: DocGen |},
+|}): Node {
+  return (
+    <Page title="RadioGroup">
+      <PageHeader
+        name="RadioGroup"
+        description={generatedDocGen.RadioGroup?.description}
+        defaultCode={`
+      function RadioButtonExample() {
+        const [favorite, setFavorite] = React.useState(undefined);
+
+        return (
+          <RadioGroup legend="Gender">
+            <RadioGroup.RadioButton
+              checked={favorite === 'Female'}
+              id="genderFemale"
+              label="Female"
+              name="gender-pref"
+              onChange={() => setFavorite( 'Female' )}
+              value="Female"
+            />
+            <RadioGroup.RadioButton
+              checked={favorite === 'Male'}
+              id="genderMale"
+              label="Male"
+              name="gender-pref"
+              onChange={() => setFavorite( 'Male' )}
+              value="Male"
+            />
+            <RadioGroup.RadioButton
+              checked={favorite === 'Non-binary'}
+              id="genderNon-binary"
+              label="Non-binary"
+              name="gender-pref"
+              onChange={() => setFavorite( 'Non-binary' )}
+              value="Non-binary"
+            />
+            <RadioGroup.RadioButton
+              checked={favorite === 'Prefer not to state'}
+              id="genderPrefer not to state"
+              label="Prefer not to state"
+              name="gender-pref"
+              onChange={() => setFavorite( 'Prefer not to state' )}
+              value="Prefer not to state"
+            />
+          </RadioGroup>
+        );
+      }
+      `}
+      />
+
+      <GeneratedPropTable generatedDocGen={generatedDocGen.RadioGroup} />
+      <GeneratedPropTable
+        Component={RadioGroup?.RadioButton}
+        name="RadioGroup.RadioButton"
+        id="RadioGroup.RadioButton"
+        generatedDocGen={generatedDocGen.RadioGroupButton}
+      />
+
+      <MainSection name="Usage guidelines">
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            cardSize="md"
+            type="do"
+            title="When to use"
+            description={`
+          - In a list, form or a Table, to present users with multiple, related options where only one option can be selected.
```

Comment:
The table usage can be a bit more abstract than just our component specifically, so more like "In a list, form, or table, to present…"